scripts/gas-refund-program/utils.ts

```diff
@@ -78,78 +78,61 @@ export const startOfDayMilliSec = (timestamp: number) => {
   return Math.floor(timestamp / DAY_SEC_MSEC) * DAY_SEC_MSEC;
 };
 
-export const generateHourlyTimestamps = (
-  startUnixTimestamp: number,
-  endUnixTimestamp: number,
-) => {
-  const startOfHourTimestampUnix = startOfHourSec(startUnixTimestamp);
-  const endOfHourTimestampUnix = startOfHourSec(endUnixTimestamp);
-  const hoursInBetween = Math.floor(
-    (endOfHourTimestampUnix - startOfHourTimestampUnix) / ONE_HOUR_SEC,
-  );
-  const hourlyTimestamps = _.range(0, hoursInBetween + 1).map(
-    i => startOfHourTimestampUnix + i * ONE_HOUR_SEC,
-  );
-
-  return hourlyTimestamps;
-};
-
-async function fetchBlockTimestamp({
+async function fetchBlocksTimestamps({
   chainId,
-  blockNumber,
+  blockNumbers: _blockNumbers,
 }: {
   chainId: number;
-  blockNumber: number;
-}): Promise<number> {
-  const subgraphURL = SUBGRAPH_URL[chainId];
-  const query = `query ($block: BigInt) {
-      blocks(first: 1, where: {number: $block}) {
+  blockNumbers: number[];
+}): Promise<{ [blockNumber: number]: number }> {
+  const blockNumbers = _.uniq(_blockNumbers);
+
+  const sliceLength = 100;
+
+  const execute = async (
+    blockNumberSliced: number[],
+  ): Promise<[{ number: string; timestamp: string }]> => {
+    const subgraphURL = SUBGRAPH_URL[chainId];
+    const query = `query ($sliceLength: Int, $blocks: [BigInt!]!) {
+      blocks(first: 100, where: {number_in: $blocks}) {
```

Comment:
did you mean to use $sliceLength for `first`? otherwise I'm not sure why the `sliceLength` param is pased below in `variables`.

Reply:
yes exactly I forgot to replace the var thx for reporting